src/renderer/widgets/Label.cpp

```diff
@@ -111,12 +112,15 @@ bool CLabel::draw(const SRenderData& data) {
             asset             = newAsset;
             resourceID        = pendingResourceID;
             pendingResourceID = "";
-            pos               = posFromHVAlign(viewport, asset->texture.m_vSize, configPos, halign, valign);
             shadow.markShadowDirty();
         }
     }
 
+    // calc pos
+    pos = posFromHVAlign(viewport, asset->texture.m_vSize, configPos, halign, valign, angle);
+
     CBox box = {pos.x, pos.y, asset->texture.m_vSize.x, asset->texture.m_vSize.y};
+    box.rot  = angle;
     shadow.draw(data);
```

Comment:
dont you need to pass rot to shadow too?

Reply:
mmm not sure, seems all apply correctly like so, same as if I move it upper

![image](https://github.com/hyprwm/hyprlock/assets/130279855/22b5ccc6-4587-4446-affa-a251a950fdd9)


do it?

Reply:
if it works it works whatever